src/message-handler/src/NetworkDeserializer.cpp

```diff
@@ -0,0 +1,25 @@
+#include "NetworkDeserializer.h"
+#include <pb_decode.h>
+
+NetworkDeserializer::NetworkDeserializer(INetworkInputStream& inputStream,
+                                         INetworkManager& networkManager) :
+    m_inputStream(inputStream), m_networkManager(networkManager) {}
+
+bool NetworkDeserializer::deserializeFromStream(MessageDTO& message) {
```

Comment:
The HiveMindHostDeserializer expects an IProtobufStream to be injected but my unicast communication is split by input/output. I could change it in Propolis to be injected an IProtobufInputStream, it'll be cleaner. And since I don't use the manager i'll remove it and only add it when necessary